RecoEgamma/EgammaHLTProducers/src/EgammaHLTBcHcalIsolationProducersRegional.cc

```diff
@@ -119,17 +138,23 @@ void EgammaHLTBcHcalIsolationProducersRegional::produce(edm::Event& iEvent, cons
       else
 	isol = isolAlgo.getTowerEtSum(&(*recoEcalCandRef));
 
-      if (doRhoCorrection_) {
-	if (fabs(recoEcalCandRef->superCluster()->eta()) < 1.442)
-	  isol = isol - rho*effectiveAreaBarrel_;
-	else
-	  isol = isol - rho*effectiveAreaEndcap_;
-      }
     } else { //calcuate H for H/E
       if (useSingleTower_)
 	isol = hcalHelper_->hcalESumDepth1BehindClusters(towersBehindCluster) + hcalHelper_->hcalESumDepth2BehindClusters(towersBehindCluster);
       else
 	isol = hcalHelper_->hcalESum(*(recoEcalCandRef->superCluster()));
+
+    }
+
+    if (doRhoCorrection_) {
+      int iEA = -1;
+      for (int bIt = absEtaLowEdges_.size() - 1; bIt > -1; bIt--) {
+        if ( fabs(recoEcalCandRef->superCluster()->eta()) > absEtaLowEdges_.at(bIt) ) {
```

Comment:
> On Aug 23, 2016, at 2:58 PM, Andrea Bocci <notifications@github.com> wrote:
> 
> In RecoEgamma/EgammaHLTProducers/src/EgammaHLTBcHcalIsolationProducersRegional.cc:
> 
> >      } else { //calcuate H for H/E
> >        if (useSingleTower_)
> >  	isol = hcalHelper_->hcalESumDepth1BehindClusters(towersBehindCluster) + hcalHelper_->hcalESumDepth2BehindClusters(towersBehindCluster);
> >        else
> >  	isol = hcalHelper_->hcalESum(*(recoEcalCandRef->superCluster()));
> > +
> > +    }
> > +
> > +    if (doRhoCorrection_) {
> > +      int iEA = -1;
> > +      for (int bIt = absEtaLowEdges_.size() - 1; bIt > -1; bIt--) {
> > +        if ( fabs(recoEcalCandRef->superCluster()->eta()) > absEtaLowEdges_.at(bIt) ) {
> 
> @davidlange6 I do I agree with you, but I'm wondering - is fabs really an expensive operation ? It should be just a bitwise "and", no ?
> 

indeed, I tried a test.. either the compiler we have optimizes this away or its very cheap in this case.. good to know

> Anyway, @afiqaize , please do make the change. At the same time, it is better to use auto eta = std::abs(...) rather than fabs(...) to automatically pick up the correct precision.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>